File: ruby/lib/runapi/openai_tts/resources/text_to_speech.rb
# frozen_string_literal: true

module RunApi
  module OpenaiTts
    module Resources
      # Generates RunAPI-managed MP3 speech from text.
      class TextToSpeech
        include RunApi::Core::ResourceHelpers

        ENDPOINT = "/api/v1/openai_tts/text_to_speech"
        RESPONSE_CLASS = Types::TextToSpeechResponse

        def initialize(http)
          @http = http
        end

        def run(options: nil, **params)
          params = compact_params(params)
          validate_contract!(CONTRACT["text-to-speech"], params)
          request(:post, ENDPOINT, body: params, options: options)
        end
      end
    end
  end
end

File: python/src/runapi/openai_tts/resources/text_to_speech.py
"""OpenAI TTS text-to-speech resource."""

from __future__ import annotations

from typing import Any, Optional

from runapi.core import RequestOptions, Resource

from ..contract_gen import CONTRACT
from ..types import TextToSpeechResponse


class TextToSpeech(Resource):
    """Generate a RunAPI-managed MP3 from text."""

    ENDPOINT = "/api/v1/openai_tts/text_to_speech"
    RESPONSE_CLASS = TextToSpeechResponse

    def run(self, options: Optional[RequestOptions] = None, **params: Any) -> Any:
        compacted = self._compact_params(params)
        self._validate_contract(CONTRACT["text-to-speech"], compacted)
        return self._request("post", self.ENDPOINT, body=compacted, options=options)
